# This file is part of HiGFXback

# requires
REQUIRES="autotools-wrappers-lt-build glib-build jpeg-build libpng-build tiff-build"

pkg-config --exists --print-errors $REQUIRES || exit 1

# configure (ac-2.69; am-1.15; lt-2.4.6)
libtoolize -c -f; aclocal -I m4; autoheader; autoconf; automake -a -c
./configure --prefix=/drm

# build
make

# install
install -d $DESTDIR/drm/bin
install gdk-pixbuf/.libs/gdk-pixbuf-csource $DESTDIR/drm/bin
install gdk-pixbuf/.libs/gdk-pixbuf-pixdata $DESTDIR/drm/bin
install gdk-pixbuf/.libs/gdk-pixbuf-query-loaders $DESTDIR/drm/bin
install thumbnailer/.libs/gdk-pixbuf-thumbnailer $DESTDIR/drm/bin
install -d $DESTDIR/drm/include/gdk-pixbuf-2.0/gdk-pixbuf
install -m 644 gdk-pixbuf/gdk-pixbuf.h $DESTDIR/drm/include/gdk-pixbuf-2.0/gdk-pixbuf
install -m 644 gdk-pixbuf/gdk-pixbuf-animation.h $DESTDIR/drm/include/gdk-pixbuf-2.0/gdk-pixbuf
install -m 644 gdk-pixbuf/gdk-pixbuf-autocleanups.h $DESTDIR/drm/include/gdk-pixbuf-2.0/gdk-pixbuf
install -m 644 gdk-pixbuf/gdk-pixbuf-core.h $DESTDIR/drm/include/gdk-pixbuf-2.0/gdk-pixbuf
install -m 644 gdk-pixbuf/gdk-pixbuf-enum-types.h $DESTDIR/drm/include/gdk-pixbuf-2.0/gdk-pixbuf
install -m 644 gdk-pixbuf/gdk-pixbuf-features.h $DESTDIR/drm/include/gdk-pixbuf-2.0/gdk-pixbuf
install -m 644 gdk-pixbuf/gdk-pixbuf-io.h $DESTDIR/drm/include/gdk-pixbuf-2.0/gdk-pixbuf
install -m 644 gdk-pixbuf/gdk-pixbuf-loader.h $DESTDIR/drm/include/gdk-pixbuf-2.0/gdk-pixbuf
install -m 644 gdk-pixbuf/gdk-pixbuf-macros.h $DESTDIR/drm/include/gdk-pixbuf-2.0/gdk-pixbuf
install -m 644 gdk-pixbuf/gdk-pixbuf-simple-anim.h $DESTDIR/drm/include/gdk-pixbuf-2.0/gdk-pixbuf
install -m 644 gdk-pixbuf/gdk-pixbuf-transform.h $DESTDIR/drm/include/gdk-pixbuf-2.0/gdk-pixbuf
install -m 644 gdk-pixbuf/gdk-pixdata.h $DESTDIR/drm/include/gdk-pixbuf-2.0/gdk-pixbuf
install -d $DESTDIR/drm/lib
install gdk-pixbuf/.libs/libgdk_pixbuf-2.0.so.0 $DESTDIR/drm/lib
ln -sf libgdk_pixbuf-2.0.so.0 $DESTDIR/drm/lib/libgdk_pixbuf-2.0.so
install -d $DESTDIR/drm/lib/gdk-pixbuf-2.0/2.10.0/loaders
install gdk-pixbuf/.libs/libpixbufloader-ani.so $DESTDIR/drm/lib/gdk-pixbuf-2.0/2.10.0/loaders
install gdk-pixbuf/.libs/libpixbufloader-bmp.so $DESTDIR/drm/lib/gdk-pixbuf-2.0/2.10.0/loaders
install gdk-pixbuf/.libs/libpixbufloader-gif.so $DESTDIR/drm/lib/gdk-pixbuf-2.0/2.10.0/loaders
install gdk-pixbuf/.libs/libpixbufloader-icns.so $DESTDIR/drm/lib/gdk-pixbuf-2.0/2.10.0/loaders
install gdk-pixbuf/.libs/libpixbufloader-ico.so $DESTDIR/drm/lib/gdk-pixbuf-2.0/2.10.0/loaders
install gdk-pixbuf/.libs/libpixbufloader-jpeg.so $DESTDIR/drm/lib/gdk-pixbuf-2.0/2.10.0/loaders
install gdk-pixbuf/.libs/libpixbufloader-png.so $DESTDIR/drm/lib/gdk-pixbuf-2.0/2.10.0/loaders
install gdk-pixbuf/.libs/libpixbufloader-pnm.so $DESTDIR/drm/lib/gdk-pixbuf-2.0/2.10.0/loaders
install gdk-pixbuf/.libs/libpixbufloader-qtif.so $DESTDIR/drm/lib/gdk-pixbuf-2.0/2.10.0/loaders
install gdk-pixbuf/.libs/libpixbufloader-tga.so $DESTDIR/drm/lib/gdk-pixbuf-2.0/2.10.0/loaders
install gdk-pixbuf/.libs/libpixbufloader-tiff.so $DESTDIR/drm/lib/gdk-pixbuf-2.0/2.10.0/loaders
install gdk-pixbuf/.libs/libpixbufloader-xbm.so $DESTDIR/drm/lib/gdk-pixbuf-2.0/2.10.0/loaders
install gdk-pixbuf/.libs/libpixbufloader-xpm.so $DESTDIR/drm/lib/gdk-pixbuf-2.0/2.10.0/loaders
install -d $DESTDIR/drm/lib/pkgconfig
install -m 644 gdk-pixbuf-2.0.pc $DESTDIR/drm/lib/pkgconfig
LD_LIBRARY_PATH=$DESTDIR/drm/lib $DESTDIR/drm/bin/gdk-pixbuf-query-loaders > $DESTDIR/drm/lib/gdk-pixbuf-2.0/2.10.0/loaders.cache

# build.pc
install -d $DESTDIR/drm/share/pkgconfig
cat > $DESTDIR/drm/share/pkgconfig/gdk-pixbuf-build.pc << EOF
Name: gdk-pixbuf
Version: 2.36.12
Description: Image loading library
Requires: $REQUIRES

devel=\\
/drm/include/gdk-pixbuf-2.0/gdk-pixbuf/gdk-pixbuf.h \\
/drm/include/gdk-pixbuf-2.0/gdk-pixbuf/gdk-pixbuf-animation.h \\
/drm/include/gdk-pixbuf-2.0/gdk-pixbuf/gdk-pixbuf-core.h \\
/drm/include/gdk-pixbuf-2.0/gdk-pixbuf/gdk-pixbuf-enum-types.h \\
/drm/include/gdk-pixbuf-2.0/gdk-pixbuf/gdk-pixbuf-features.h \\
/drm/include/gdk-pixbuf-2.0/gdk-pixbuf/gdk-pixbuf-io.h \\
/drm/include/gdk-pixbuf-2.0/gdk-pixbuf/gdk-pixbuf-loader.h \\
/drm/include/gdk-pixbuf-2.0/gdk-pixbuf/gdk-pixbuf-macros.h \\
/drm/include/gdk-pixbuf-2.0/gdk-pixbuf/gdk-pixbuf-simple-anim.h \\
/drm/include/gdk-pixbuf-2.0/gdk-pixbuf/gdk-pixbuf-transform.h \\
/drm/include/gdk-pixbuf-2.0/gdk-pixbuf/gdk-pixdata.h \\
/drm/lib/libgdk_pixbuf-2.0.so \\
/drm/lib/pkgconfig/gdk-pixbuf-2.0.pc

exec=\\
/drm/bin/gdk-pixbuf-csource \\
/drm/bin/gdk-pixbuf-pixdata \\
/drm/bin/gdk-pixbuf-query-loaders \\
/drm/bin/gdk-pixbuf-thumbnailer \\
/drm/lib/libgdk_pixbuf-2.0.so.0 \\
/drm/lib/gdk-pixbuf-2.0/2.10.0/loaders.cache \\
/drm/lib/gdk-pixbuf-2.0/2.10.0/loaders/libpixbufloader-ani.so \\
/drm/lib/gdk-pixbuf-2.0/2.10.0/loaders/libpixbufloader-bmp.so \\
/drm/lib/gdk-pixbuf-2.0/2.10.0/loaders/libpixbufloader-gif.so \\
/drm/lib/gdk-pixbuf-2.0/2.10.0/loaders/libpixbufloader-icns.so \\
/drm/lib/gdk-pixbuf-2.0/2.10.0/loaders/libpixbufloader-ico.so \\
/drm/lib/gdk-pixbuf-2.0/2.10.0/loaders/libpixbufloader-jpeg.so \\
/drm/lib/gdk-pixbuf-2.0/2.10.0/loaders/libpixbufloader-png.so \\
/drm/lib/gdk-pixbuf-2.0/2.10.0/loaders/libpixbufloader-pnm.so \\
/drm/lib/gdk-pixbuf-2.0/2.10.0/loaders/libpixbufloader-qtif.so \\
/drm/lib/gdk-pixbuf-2.0/2.10.0/loaders/libpixbufloader-tga.so \\
/drm/lib/gdk-pixbuf-2.0/2.10.0/loaders/libpixbufloader-tiff.so \\
/drm/lib/gdk-pixbuf-2.0/2.10.0/loaders/libpixbufloader-xbm.so \\
/drm/lib/gdk-pixbuf-2.0/2.10.0/loaders/libpixbufloader-xpm.so
EOF

exit

# patch
--- gdk-pixbuf-2.36.12.orig/configure.ac
+++ gdk-pixbuf-2.36.12/configure.ac
@@ -40,7 +40,7 @@
 cflags_set=${CFLAGS+set}
 
 AM_INIT_AUTOMAKE([1.11 foreign no-define -Wno-portability dist-xz no-dist-gzip tar-ustar])
-AM_MAINTAINER_MODE([enable])
+AM_MAINTAINER_MODE
 
 # Support silent build rules. Disable
 # by either passing --disable-silent-rules to configure or passing V=0
@@ -102,6 +102,11 @@
 AC_CHECK_LIB(m, round, AC_DEFINE(HAVE_ROUND, [1], [Define to 1 if libm has round]))
 AC_CHECK_LIB(m, lrint, AC_DEFINE(HAVE_LRINT, [1], [Define to 1 if libm has lrint]))
 
+AC_ARG_ENABLE(gio-sniffing,
+              [AC_HELP_STRING([--enable-gio-sniffing=@<:@no/yes/auto@:>@],
+                              [enable or disable gio sniffing [default=auto]])],,
+              [enable_gio_sniffing=auto])
+
 AC_MSG_CHECKING([for native Win32])
 LIB_EXE_MACHINE_FLAG=X86
 EXE_MANIFEST_ARCHITECTURE=X86
@@ -670,10 +675,55 @@
 # check one of the variables here
 AM_CONDITIONAL(INCLUDE_GDIPLUS, [test x"$INCLUDE_gdip_ico" = xyes])
 
-if test x$gio_can_sniff = x; then
-  # Will not run on win32, so require shared-mime-info
-  PKG_CHECK_MODULES(SHARED_MIME_INFO, shared-mime-info)
-  AC_DEFINE(GDK_PIXBUF_USE_GIO_MIME, 1, [Define if gio can sniff image data])
+if test x$enable_gio_sniffing = xauto && test x$gio_can_sniff = x; then
+  AC_MSG_CHECKING([if gio can sniff png])
+  gtk_save_LIBS="$LIBS"
+  gtk_save_CFLAGS="$CFLAGS"
+  LIBS="`$PKG_CONFIG --libs gio-2.0`"
+  CFLAGS="`$PKG_CONFIG --cflags gio-2.0`"
+  AC_RUN_IFELSE([AC_LANG_SOURCE([[
+  #include <gio/gio.h>
+  static const gsize data_size = 159;
+  static const guint8 data[] =
+  {
+    0x89, 0x50, 0x4e, 0x47, 0x0d, 0x0a, 0x1a, 0x0a, 0x00, 0x00, 0x00, 0x0d,
+    0x49, 0x48, 0x44, 0x52, 0x00, 0x00, 0x00, 0x01, 0x00, 0x00, 0x00, 0x01,
+    0x08, 0x02, 0x00, 0x00, 0x00, 0x90, 0x77, 0x53, 0xde, 0x00, 0x00, 0x00,
+    0x01, 0x73, 0x52, 0x47, 0x42, 0x00, 0xae, 0xce, 0x1c, 0xe9, 0x00, 0x00,
+    0x00, 0x09, 0x70, 0x48, 0x59, 0x73, 0x00, 0x00, 0x0b, 0x13, 0x00, 0x00,
+    0x0b, 0x13, 0x01, 0x00, 0x9a, 0x9c, 0x18, 0x00, 0x00, 0x00, 0x07, 0x74,
+    0x49, 0x4d, 0x45, 0x07, 0xd8, 0x07, 0x0f, 0x10, 0x08, 0x15, 0x61, 0xd8,
+    0x35, 0x37, 0x00, 0x00, 0x00, 0x19, 0x74, 0x45, 0x58, 0x74, 0x43, 0x6f,
+    0x6d, 0x6d, 0x65, 0x6e, 0x74, 0x00, 0x43, 0x72, 0x65, 0x61, 0x74, 0x65,
+    0x64, 0x20, 0x77, 0x69, 0x74, 0x68, 0x20, 0x47, 0x49, 0x4d, 0x50, 0x57,
+    0x81, 0x0e, 0x17, 0x00, 0x00, 0x00, 0x0c, 0x49, 0x44, 0x41, 0x54, 0x08,
+    0xd7, 0x63, 0xf8, 0xff, 0xff, 0x3f, 0x00, 0x05, 0xfe, 0x02, 0xfe, 0xdc,
+    0xcc, 0x59, 0xe7, 0x00, 0x00, 0x00, 0x00, 0x49, 0x45, 0x4e, 0x44, 0xae,
+    0x42, 0x60, 0x82
+  };
+  int
+  main (int argc, char **argv)
+  {
+    char *content_type;
+    char *image_png;
+    g_type_init ();
+    content_type = g_content_type_guess (NULL, data, data_size, NULL);
+    image_png = g_content_type_from_mime_type ("image/png");
+    return !!g_strcmp0 (content_type, image_png);
+  }]])],
+    [gio_can_sniff=yes
+     AC_DEFINE(GDK_PIXBUF_USE_GIO_MIME, 1, [Define if gio can sniff image data])],
+    [gio_can_sniff=no])
+  AC_MSG_RESULT($gio_can_sniff)
+  LIBS="$gtk_save_LIBS"
+  CFLAGS="$gtk_save_CFLAGS"
+else
+  if test x$enable_gio_sniffing = xyes; then
+     gio_can_sniff=yes
+     AC_DEFINE(GDK_PIXBUF_USE_GIO_MIME, 1, [Define if gio can sniff image data])
+  else
+     gio_can_sniff=no
+  fi
 fi
 
 AM_CONDITIONAL(HAVE_TIFF, test "x$libtiff_found" != xno)
--- gdk-pixbuf-2.36.12.orig/gdk-pixbuf-2.0.pc.in
+++ gdk-pixbuf-2.36.12/gdk-pixbuf-2.0.pc.in
@@ -7,6 +7,7 @@
 gdk_pixbuf_binarydir=@libdir@/gdk-pixbuf-@GDK_PIXBUF_API_VERSION@/@GDK_PIXBUF_BINARY_VERSION@
 gdk_pixbuf_moduledir=${gdk_pixbuf_binarydir}/loaders
 gdk_pixbuf_cache_file=${gdk_pixbuf_binarydir}/loaders.cache
+gdk_pixbuf_csource=${exec_prefix}/bin/gdk-pixbuf-csource
 
 Name: GdkPixbuf
 Description: Image loading and scaling

# source
https://download.gnome.org/sources/gdk-pixbuf/2.36/gdk-pixbuf-2.36.12.tar.xz
